### PR TITLE
CCMSPUI-376: Migrate notification summary from SOA-API to EBS-API

### PR DESCRIPTION
**Pull request by JamieBriggs-MoJ**
As part of CBP-947, work was done to move the notification summary from SOA to EBS, which resulted in a new view `XXCCMS_NOTIFICATION_COUNT_V` being created to support this. 

As part of this PR and CCMSPUI-376, a new endpoint has been created which uses the new view.

### CODE REVIEW

**Review by farrell-m — COMMENTED** [review]
Looks good. Might just be personal preference, but I think notifications would sit better under their own controller/service, as those used for User I see as more for user metadata and user management.

**Review by PhilDigitalJustice — COMMENTED** [review]
1 minor comment

File: data-service/src/main/java/uk/gov/laa/ccms/data/entity/NotificationCount.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    /**
+     * Represents the type of notification associated with a user login session.
+     * This field is mapped to the "NOTIFICATION_TYPE" column in the database and
+     * uses the NotificationTypeConverter to translate between the NotificationType
+     * enumeration and its corresponding String representation for database storage.
+     */
+    @Column(name = "NOTIFICATION_TYPE")
+    @Convert(converter = NotificationTypeConverter.class)
+    private NotificationType notificationType;
+  }
```
[the remark this answers is not in the document]
Comment (JamieBriggs-MoJ) [reply data-service/src/main/java/uk/gov/laa/ccms/data/entity/NotificationCount.java]: Good point, I have moved this into it's own class for consistency now

**Comment by JamieBriggs-MoJ** [comment]
> Looks good. Might just be personal preference, but I think notifications would sit better under their own controller/service, as those used for User I see as more for user metadata and user management.

I agree with this retrospectively. I have made some more changes to go in line with this and to decouple the notification summary stuff away from the User controller/service.

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**